enterprise/models.py

```diff
@@ -2,42 +2,238 @@
+                                         requested data sharing (learners are allowed to opt-in) or required data
+                                         sharing (learners must share their data)
+        sso_providers (CharField): list of comma-separated SSO provider slugs. Providers in this list should link
+                                    users registered with that provider with Enterprise Customer
+        active (BooleanField): used to mark inactive Enterprise Customers - implements "soft delete" pattern
     """
 
-    # TODO: add field definitions
+    class Meta(object):
+        """
+        Meta class for EnterpriseCustomerHistory model.
+        """
+
+        app_label = "enterprise"
+        verbose_name = "Enterprise Customer History"
+        verbose_name_plural = "Enterprise Customer Histories"
+
+    origin = models.ForeignKey(EnterpriseCustomer, related_name="history")
+    name = models.CharField(max_length=500, blank=False, null=False, help_text=_("Enterprise Customer name."))
+    short_code = models.CharField(
+        max_length=120, blank=False, null=False,
+        help_text=_("Short code will be used to reference this Enterprise Customer in other parts of the system,"
+                    "i.e. SSO, ecommerce app, analytics etc.")
+    )
+    logo = models.ImageField(
+        upload_to="enterprise_logos",
+        help_text=_("Please add only .PNG files for logo images."),
+        null=True, blank=True, max_length=255
+    )
+    data_sharing_policy = models.CharField(
+        blank=False, null=False, help_text=_("Data sharing policy"), max_length=20,
+        choices=(
+            (EnterpriseCustomer.NO_DATA_SHARING, _("No data sharing requested")),
+            (EnterpriseCustomer.REQUEST_DATA_SHARING, _("Request data sharing")),
+            (EnterpriseCustomer.REQUIRE_DATA_SHARING, _("Require data sharing")),
+        )
+    )
+    active = models.BooleanField(default=True)
 
     def __str__(self):
         """
-        Get a string representation of this model instance.
+        Return human-readable string representation.
+        """
+        return "<EnterpriseCustomerHistory {ID}: {name}, Last Modified: {modified}>".format(
+            ID=self.id,
+            modified=self.modified,
+            name=self.name,
+        )
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__unicode__()
+
+
+@python_2_unicode_compatible
+class EnterpriseCustomerMember(models.Model):
+    """
+    Model that keeps track of user - enterprise customer affinity.
+
+    Fields:
+        enterprise_customer (ForeignKey[EnterpriseCustomer]): enterprise customer
+        user_id (IntegerField): user identifier
+    """
+
+    enterprise_customer = models.ForeignKey(EnterpriseCustomer, blank=False, null=False)
+    user_id = models.IntegerField(null=False, blank=False)
```

Comment:
@mtyaka because we don't want to tie enterprise to anything in LMS, as there are plans to make it a separate service/app in the future. We've discussed that with @mattdrayer on HipChat on Friday and decided that storing user_ids instead of foreign key will allow for greater flexibility in future, while also not significantly complicating current setup.